pkg/cmd/cli/cmd/newapp.go

```diff
@@ -593,7 +593,7 @@ func retryBuildConfig(info *resource.Info, err error) runtime.Object {
 		buildapi.GenericWebHookBuildTriggerType: {},
 		buildapi.ImageChangeBuildTriggerType:    {},
 	}
-	if info.Mapping.GroupVersionKind.GroupKind() == buildapi.Kind("BuildConfig") && isInvalidTriggerError(err) {
+	if info.Mapping.GroupVersionKind.GroupKind() == buildapi.Kind("BuildConfig") || info.Mapping.GroupVersionKind.GroupKind() == buildapi.LegacyKind("BuildConfig") && isInvalidTriggerError(err) {
```

Comment:
I'm not sure this is returning what we want. I'd prefer explicit parentheses around the GroupKind check, like:
`(Kind || LegacyKind) && ...`